### ISSUE TITLE
[Feature request] Muon-Prodigy hybrid for diffusion training speedup

### ISSUE THREAD

**Issue by kabachuha** [tdrussell/diffusion-pipe#336]
Hi!

There is an awesome idea: bring Prodigy's adaptive learning rate into Muon to combine Prodigy's ease of use with Muon's training stability.

I think it should be easy to (vibe) code (In fact, I already did some experiments on toy diffusion datasets, and it's a new level of quality - will share later) and I'm sure that it will bring great results.

**Comment by ghost** [comment]
Have you tried automagic? I found it quite good.

**Comment by kabachuha** [comment]
It is too chaotic to me and unstable

**Comment by ghost** [comment]
> It is too chaotic to me and unstable

For me its been the best optimizer ive used and has made my training stable for wan 2.1 1.3b it removed the need for me to manually tune the LR to get good results. Massive time saver.

**Comment by feeziwater** [comment]
How did your Muon-Prodigy hybrid output turn out?
Would you mind sharing the output and code?

**Comment by kabachuha** [comment]
Yes, this weekend I think. I got irl busy for the time being

**Comment by feeziwater** [comment]
I can't wait to try that out!

**Comment by haoxu2000** [comment]
is there any news? Muon-Prodigy hybrid sounds nice.

**Comment by feeziwater** [comment]
> is there any news? Muon-Prodigy hybrid sounds nice.

I tried the vibe code myself with the Muon-Prodigy optimizer using the two official repositories.

I think it's working, but I haven't been able to find a good configuration for it.

It seems to overfit too early compared to AdamW.

So I'm curious how he managed to achieve new level of quality.

**Comment by kabachuha** [comment]
Yes, I tested it now, and while it looks cool for toy datasets and gives interesting pictures, it's compares worse to well scheduled AdamW/Prodigy, so I think it's not really practical :(
